File: python/maplestory_openapi/api/dto/character/character_propensity.py
from datetime import datetime
from pydantic import BaseModel


class CharacterPropensity(BaseModel):
    """캐릭터 성향 정보

    date(datetime): 조회 기준일
    charisma_level(int): 카리스마 레벨
    sensibility_level(int): 감성 레벨
    insight_level(int): 통찰력 레벨
    willingness_level(int): 의지 레벨
    handicraft_level(int): 손재주 레벨
    charm_level(int): 매력 레벨
    """
    date: datetime | None
    charisma_level: int | None
    sensibility_level: int | None
    insight_level: int | None
    willingness_level: int | None
    handicraft_level: int | None
    charm_level: int | None

File: js/src/maplestory/api/tms/dto/character/characterPropensity.ts
import * as base from '../../../common/dto/character/characterPropensity';
import { CharacterPropensityBody } from '../../response/character/characterPropensityBody';

/**
 * 角色性向資訊
 */
export class CharacterPropensityDto extends base.CharacterPropensityDto {
  /**
   * 要搜尋的日期 (TST，每日資料中的小時與分鐘將顯示為 0)
   */
  public override date: Date | null;

  /**
   * 領導力等級
   */
  public override charismaLevel: number;

  /**
   * 感性等級
   */
  public override sensibilityLevel: number;

  /**
   * 洞察力等級
   */
  public override insightLevel: number;

  /**
   * 意志等級
   */
  public override willingnessLevel: number;

  /**
   * 手藝等級
   */
  public override handicraftLevel: number;

  /**
   * 魅力等級
   */
  public override charmLevel: number;

  constructor(obj: CharacterPropensityBody) {
    super();

    const {
      date,
      charisma_level,
      sensibility_level,
      insight_level,
      willingness_level,
      handicraft_level,
      charm_level,
    } = obj;

    this.date = date ? new Date(date) : null;
    this.charismaLevel = charisma_level!;
    this.sensibilityLevel = sensibility_level!;
    this.insightLevel = insight_level!;
    this.willingnessLevel = willingness_level!;
    this.handicraftLevel = handicraft_level!;
    this.charmLevel = charm_level!;
  }
}
